src/components/Reports/FourteenMonthReports/FourteenMonthReport.tsx

```diff
@@ -0,0 +1,170 @@
+import React, { useMemo, useRef, useState } from 'react';
+import { Box, CircularProgress, Theme, useMediaQuery } from '@material-ui/core';
+import { useTranslation } from 'react-i18next';
+import { useReactToPrint } from 'react-to-print';
+import { FourteenMonthReportCurrencyType } from '../../../../graphql/types.generated';
+import { FourteenMonthReportHeader as Header } from './Layout/Header/Header';
+import type { CurrencyType } from './Layout/Header/Actions/Actions';
+import { useFourteenMonthReportQuery } from './GetFourteenMonthReport.generated';
+import type {
+  Contact,
+  Order,
+  OrderBy,
+} from './Layout/Table/TableHead/TableHead';
+import { FourteenMonthReportTable as Table } from './Layout/Table/Table';
+import { Notification } from 'src/components/Notification/Notification';
+import { EmptyReport } from 'src/components/Reports/EmptyReport/EmptyReport';
+
+interface Props {
+  accountListId: string;
+  isNavListOpen: boolean;
+  onNavListToggle: () => void;
+  title: string;
+  currencyType: CurrencyType;
+}
+
+export const FourteenMonthReport: React.FC<Props> = ({
+  accountListId,
+  currencyType,
+  isNavListOpen,
+  title,
+  onNavListToggle,
+}) => {
+  const [isExpanded, setExpanded] = useState<boolean>(false);
+  const [order, setOrder] = useState<Order>('asc');
+  const [orderBy, setOrderBy] = useState<OrderBy | number | null>(null);
+  const reportTableRef = useRef(null);
+  const { t } = useTranslation();
+
+  const isMobile = useMediaQuery((theme: Theme) =>
+    theme.breakpoints.down('sm'),
+  );
+
+  const { data, loading, error } = useFourteenMonthReportQuery({
+    variables: {
+      accountListId,
+      currencyType:
+        currencyType === 'salary'
+          ? FourteenMonthReportCurrencyType.Salary
+          : FourteenMonthReportCurrencyType.Donor,
+    },
+  });
+
+  const contacts = useMemo(() => {
+    return data?.fourteenMonthReport.currencyGroups?.flatMap(
+      (currencyGroup) => [...currencyGroup?.contacts],
+    );
+  }, [data?.fourteenMonthReport.currencyGroups]);
```

Comment:
So there are 2 main differences between the partner currency and salary currency reports.
1. The API returns numbers converted into the salary currency depending on `currencyType` - implemented
2. For partner currency, the frontend shows multiple tables grouped by partner currency. So the salary currency one is implemented correctly (combining all `currencyGroups`). But for partner currency, we want to show each `data?.fourteenMonthReport.currencyGroups` in a separate table.

Feel free to work on that in a new PR. Sorry if we didn't explain that as well as we could have.

Reply:
Makes sense to me. :)